### PR TITLE
Add a resource to manage sequences

### PR DESCRIPTION
**Pull request by dfedde**
# This is just for us at rxrevu to look at internally before I submit it back upstream. 

Adds a sequence Resource so that sequences can can be managed with terraform.

This Resource does not currently do any updates it always deletes and recreates the sequence. It does however keep the correct next_value after these updates by setting the `start` value to the previous `next_value`.

## Test Plan
* [x] acceptance tests

## References
* https://docs.snowflake.com/en/sql-reference/sql/create-sequence.html

### CODE REVIEW

**Review by jtzero — APPROVED** [review]

File: pkg/resources/sequence.go [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+	}
+
+	err := snowflake.Exec(db, sq.Create())
+	if err != nil {
+		return errors.Wrapf(err, "error creating sequence")
+	}
+
+	return ReadSequence(d, meta)
+}
+
```
Comment (jtzero) [inline pkg/resources/sequence.go:90]: missing comments?

File: pkg/resources/sequence_acceptance_test.go [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+		Steps: []resource.TestStep{
+			// CREATE
+			{
+				Config: sequenceConfig(accName, accName),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("snowflake_sequence.test_sequence", "name", accName),
+					resource.TestCheckResourceAttr("snowflake_sequence.test_sequence", "next_value", "1"),
+				),
+			},
+			// SET COMMENT AND RENAME
```
Comment (dfedde) [inline pkg/resources/sequence_acceptance_test.go]: CaSinG!!11!!1!!

File: pkg/snowflake/sequence.go [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+	return fmt.Sprintf(`ALTER SEQUENCE %v SET COMMENT = '%v'`, tb.QualifiedName(), comment)
+}
+
+func (tb *SequenceBuilder) Rename(name string) string {
+	return fmt.Sprintf(`ALTER SEQUENCE %v RENAME TO "%v"."%v"."%v"`, tb.QualifiedName(), tb.db, tb.schema, name)
+}
+
+func (tb *SequenceBuilder) Increment() string {
+	return fmt.Sprintf(`SELECT %v.nextval`, tb.QualifiedName())
+}
```
Comment (dfedde) [inline pkg/snowflake/sequence.go]: These are not used

File: pkg/resources/sequence.go [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+func ReadSequence(d *schema.ResourceData, meta interface{}) error {
```
Comment (dfedde) [inline pkg/resources/sequence.go:92]:
```suggestion
// ReadSequence implements schema.ReadFunc
func ReadSequence(d *schema.ResourceData, meta interface{}) error {
```

File: pkg/resources/sequence_acceptance_test.go (hunk 2 above)
Comment (dfedde) [reply pkg/resources/sequence_acceptance_test.go]:
```suggestion
			// Set comment and rename
```

File: pkg/resources/sequence_acceptance_test.go [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+			{
+				Config: sequenceConfigWithComment(accName, accRename, "look at me I am a comment"),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("snowflake_sequence.test_sequence", "name", accRename),
+					resource.TestCheckResourceAttr("snowflake_sequence.test_sequence", "comment", "look at me I am a comment"),
+					resource.TestCheckResourceAttr("snowflake_sequence.test_sequence", "next_value", "1"),
+				),
+			},
+			// unset comment and set increment
```
Comment (dfedde) [inline pkg/resources/sequence_acceptance_test.go]:
```suggestion
			// Unset comment and set increment
```

File: pkg/snowflake/sequence.go (hunk 3 above)
Comment (dfedde) [inline pkg/snowflake/sequence.go]:
```suggestion
```

File: pkg/snowflake/sequence_test.go [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+	r := require.New(t)
+	s := Sequence("test_sequence", "test_db", "test_schema")
+	r.Equal(`SHOW SEQUENCES LIKE 'test_sequence' IN SCHEMA "test_db"."test_schema"`, s.Show())
+}
+
+func TestSequenceIncrement(t *testing.T) {
+	r := require.New(t)
+	s := Sequence("test_sequence", "test_db", "test_schema")
+	r.Equal(`SELECT "test_db"."test_schema"."test_sequence".nextval`, s.Increment())
+}
```
Comment (dfedde) [inline pkg/snowflake/sequence_test.go]:
```suggestion
```

File: pkg/snowflake/sequence_test.go [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+	r := require.New(t)
+	s := Sequence("test_sequence", "test_db", "test_schema")
+	r.Equal(`ALTER SEQUENCE "test_db"."test_schema"."test_sequence" UNSET COMMENT`, s.RemoveComment())
+}
+
+func TestSequenceRename(t *testing.T) {
+	r := require.New(t)
+	s := Sequence("test_sequence", "test_db", "test_schema")
+	r.Equal(`ALTER SEQUENCE "test_db"."test_schema"."test_sequence" RENAME TO "test_db"."test_schema"."new_name"`, s.Rename("new_name"))
+}
```
Comment (dfedde) [inline pkg/snowflake/sequence_test.go]:
```suggestion
```

**Pull request closed without merging**